lib/student.ex

```diff
@@ -83,10 +138,13 @@ defmodule Student do
 
   @impl true
   def handle_call({:subscribe, project}, _from, student) do
+    Student.Event.dispatch(student, {:subscribed_to_project, student, project})
+
     {:reply, :ok,
      %Student{
-       name: student.name,
-       projects: [project | student.projects]
+       student
+       | name: student.name,
```

Comment:
Is this really needed? We're updating `student.name` with `student.name` :)

Reply:
agree, will remove name reassigement